test/tests/functional/pbs_test_run_count.py

```diff
@@ -123,16 +145,87 @@ def test_run_count_subjob(self):
         a = {ATTR_J: '1-2'}
         j = Job(TEST_USER, a)
         jid = self.server.submit(j)
+        self.subjob_check(jid=jid, sjid=j.create_subjob_id(jid, 1))
 
-        self.server.expect(JOB, {ATTR_state: "H", ATTR_runcount: 21},
-                           attrop=PTL_AND, id=j.create_subjob_id(jid, 1))
+    def test_run_count_subjob_in_x(self):
+        """
+        Submit a job array and check if the subjob and the parent are getting
+        held after 20 rejection from mom when there is another subjob in X
+        """
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 1},
+                            id=self.mom.shortname)
 
-        ja_comment = "Job Array Held, too many failed attempts to run subjob"
+        # Create an execjob_begin hook that rejects the job
+        self.create_reject_begin_hook()
+        time.sleep(2)
+        self.disable_reject_begin_hook()
 
-        self.server.expect(JOB, {ATTR_state: "H", ATTR_comment: (MATCH_RE,
-                           ja_comment)}, attrop=PTL_AND, id=jid)
+        a = {ATTR_J: '1-6'}
+        j = Job(TEST_USER, a)
+        j.set_sleep_time(10)
```

Comment:
Can you reduce the sleep time so that the overall test execution time would be less?

Reply:
can cause problems in slower machines

Reply:
I dont think we are accounting for such slower machines anywhere else in the framework. I guess it should be good enough with 2-3 seconds. In that case you wont be needing the following sleep of 8+7 either.

Reply:
This is not framework. There are many race conditions especially in job array test  cases that have cropped up on slow machines and have had corrections made using subsequent commits. Hence trying to be more careful here. trying to avoid race conditions. sleep of 8+7 is needed for faster machines

Reply:
There are two ways to go about this. Either you can start with lower values and re-adjust it whenever you detect a failure or you can go with an arbitrary upper limit which will never hit even in the slowest machines.
If we go with approach # 1 then need to update the sleep values whenever we observe a race condition or some other failure related to the slowness of the machine. But if you can identify the slow test machine which is the bottleneck, you can fixate the value in the beginning itself.
The approach you have chosen selects an arbitrary upper limit at the cost of test run time. Is there a reason why you have chosen 10s or is it just a random value? If every test writer chooses large sleep value, It could increase the total test execution time exponentially. Execution time is a usability feature and we should try to minimize it if we can.
In your test case the wait of 9s and 8s are done to see it the job in running state and to see it finished which is not required when you have jobs with a lower sleep time. If you still don't agree, please help me understand what problems can occur with respect to your test case in a slower machine?

Reply:
we cannot be expected to run this test on all slower machines to try converge on an accurate value (after all this code wont make it inside any rocket guidance systems). It has already been experienced in past. see PRs https://github.com/PBSPro/pbspro/pull/952 and https://github.com/PBSPro/pbspro/pull/918 where sleep time has been increased after experiencing problems on slower machines. 

Each person has definition of "large". and for me 10 seconds is not large. If, as per your definition of "large", 10 seconds is large, then please feel free to reduce it to what ever you feel is "small" before firing the tests. I wont be changing it here.